app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetController.scala

```diff
@@ -97,7 +98,11 @@ class EditSetController @Inject() (
       FieldNames.startDateYear  -> text,
       FieldNames.endDateDay     -> text,
       FieldNames.endDateMonth   -> text,
-      FieldNames.endDateYear    -> text
+      FieldNames.endDateYear    -> text,
+      FieldNames.legalStatus -> text.verifying(
+        messagesApi("edit-set.record.error.legal-status")(Lang.apply("en")),
+        value => !value.equalsIgnoreCase("ref.0")
```

Comment:
I believe the validation should be to assert that the value exists in the list of all legal statuses. Given that it's static for now, it can be done here.

Reply:
Would the validation in UI not suffice to check the value isn't empty or does the UI need to match the value wth actual the reference data?

Reply:
By UI validation do you mean the form validation here?

In any case, there seem to be two types of validation: the first is where nothing was selected (a blank), and that can covered here.

The second is where the legal status associated with the particular record somehow doesn't match any of the possible legal statuses.

What should happen in that case? I don't recall you haven't a test scenario to cover that, but I might be wrong.

Reply:
Just to know, there's a place for doing additional validation [here](https://github.com/nationalarchives/ctd-omega-editorial-frontend/blob/main/app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetController.scala#L195).

I do agree about the creators, but where we have the full list (as we will where we use a select box), we could do this validation after the normal form validation.

Reply:
I'll assume that we're not doing this second validation: ensuring that it exists in the list.

Should be safe, as (as Rob said), they select from a list.